orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationPrompt.java

```diff
@@ -0,0 +1,102 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.sap.ai.sdk.orchestration.client.model.CompletionPostRequest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Value;
+import lombok.experimental.Delegate;
+import lombok.val;
+
+/**
+ * Represents a request that can be sent to the orchestration service, containing messages and
+ * configuration for the orchestration modules. Prompts may be reused across multiple requests.
+ *
+ * @see OrchestrationClient
+ * @see OrchestrationConfig
+ */
+@Value
+@Getter(AccessLevel.PACKAGE)
+@AllArgsConstructor
+public class OrchestrationPrompt implements OrchestrationConfig<OrchestrationPrompt> {
+  @Nonnull List<Message> messages;
+  @Nonnull Map<String, String> templateParameters;
+
+  @Getter(AccessLevel.NONE)
+  @Delegate(types = IDelegate.class)
+  @Nonnull
+  DefaultOrchestrationConfig<OrchestrationPrompt> delegate =
+      DefaultOrchestrationConfig.asDelegateFor(this);
+
+  private interface IDelegate extends OrchestrationConfig<OrchestrationPrompt> {}
+
+  /**
+   * Initialize a prompt with the given user message.
+   *
+   * @param message A user message.
+   */
+  public OrchestrationPrompt(@Nonnull final String message) {
+    this(List.of(new UserMessage(message)), Map.of());
+  }
+
+  /**
+   * Initialize a prompt from the given messages.
+   *
+   * @param message The first message.
+   * @param messages Optionally, more messages.
+   */
+  public OrchestrationPrompt(@Nonnull final Message message, @Nullable final Message... messages) {
```

Comment:
(Minor)

AFAIK `@Nullable` varargs only makes sense if you expect an explicitly invoked `new OrchestrationPrompt(msg, (Message[]) null)`, which I wouldn't find acceptable. Empty varargs are handled as empty array. Please correct me if I'm wrong here.

Reply:
You are completely right, `@Nullable` doesn't make sense. Fixed it here and also in other places I used this pattern.